pymongo/mongo_client.py

```diff
@@ -830,6 +833,10 @@ def target():
         self_ref: Any = weakref.ref(self, executor.close)
         self._kill_cursors_executor = executor
 
+        # Add this client to the list of weakly referenced items.
+        # This will be used later if we fork.
+        MongoClient._clients.add(self)
```

Comment:
Let's move this to be the last thing in `__init__`. Also is WeakSet.add thread safe?

Reply:
I don't think it's thread-safe for `add`s. For iteration, `WeakSet` uses `_IterationGuard` to protect items in the set, so I think only `add` might need a lock.